apps/site/util/detectOS.ts

```diff
@@ -1,21 +1,16 @@
 import type { UserOS } from '@/types/userOS';
 
 export const detectOsInUserAgent = (userAgent: string | undefined): UserOS => {
-  const osMatch = userAgent?.match(/(Win|Mac|Linux)/);
-  switch (osMatch && osMatch[1]) {
-    case 'Win':
-      return 'WIN';
-    case 'Mac':
-      return 'MAC';
-    case 'Linux':
-      return 'LINUX';
-    case 'AIX':
-      return 'AIX';
-    default:
-      return 'OTHER';
-  }
+  const osMatch = userAgent?.match(/(Win|Mac|Linux|AIX)/);
+  const osMap: Record<string, UserOS> = {
+    Win: 'WIN',
+    Mac: 'MAC',
+    Linux: 'LINUX',
+    AIX: 'AIX',
+  };
+  return osMap[osMatch?.[1] ?? ''] || 'OTHER';
```

Comment:
if we are already matching, why not just use toUpperCase if we find a match, instead of a map?

Reply:
1 sec Ill be doing that rn